### ISSUE TITLE
Unresolved name of argument in default value

### ISSUE THREAD

**Issue by clebert** [facebook/flow#1127]
```javascript
/* @flow */

type Middleware = (ctx: Object, next: () => Promise) => Promise

export function createNamedMiddleware (middleware: Middleware, name: string = middleware.name): Middleware {
  return async function namedMiddleware (ctx, next) {
    // ...
  }
}
```

```shell
flow check

src/middlewares/named.js:5
  5: export function createNamedMiddleware (middleware: Middleware, name: string = middleware.name): Middleware {
                                                                                   ^^^^^^^^^^ identifier `middleware`. Could not resolve name


Found 1 error
```

**Comment by jeffmo** [comment]
This was fixed by #872, which should be making its way out in the next release (which coming very soon, but possibly delayed a bit by the holiday week)

**Comment by jeffmo** [comment]
PS: Thanks for the report! :)
